src/dsql/DdlNodes.epp

```diff
@@ -2941,7 +2941,7 @@ bool CreateAlterProcedureNode::executeAlter(thread_db* tdbb, DsqlCompilerScratch
 				if (secondPass)
 				{
 					P.RDB$PROCEDURE_TYPE.NULL = FALSE;
-					P.RDB$PROCEDURE_TYPE = (USHORT) prc_selectable;
+					P.RDB$PROCEDURE_TYPE = returns.getCount() != 0 ? (USHORT) prc_selectable : (USHORT) prc_executable;
```

Comment:
How this change is related with PR purposes ?

Reply:
> How this change is related with PR purposes ?

In fact, it's unrelated. Should I delete this change?

Reply:
A remainder of the older patch. AFAIU, it should not be necessary once external routines are extracted in one pass. @vasiliy-yashkov do you agree?

Reply:
Sure. If it is a part of some other bug fix - feel free to create specific PR.

Reply:
> A remainder of the older patch. AFAIU, it should not be necessary once external routines are extracted in one pass. @vasiliy-yashkov do you agree?

I agree.

Reply:
> Sure. If it is a part of some other bug fix - feel free to create specific PR.

Ok :)